### PR TITLE
davinci-resolve: 17.4.3 -> 18.1.4

### PR DESCRIPTION
**Pull request by yuuyins**
###### Description of changes

See https://github.com/NixOS/nixpkgs/issues/238026#issuecomment-1613164700, https://github.com/NixOS/nixpkgs/issues/238026#issuecomment-1613715624

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by BKSalman** [comment]
> @BKSalman @icodeforyou-dot-net @jshcmpbll @davidak 
> 
> 
> 
> please review

Thank you for taking the time and working on this!! However unfortunately I'm in a vacation right now, and don't have my machine, so I can't review it now, but I think it should be all good as long as you used deps from the gist, and you verified it yourself, maybe the other dear commenters can review it!

**Review by l0b0 — APPROVED** [review]
I don't know the program, but the PR looks great - alphabetically ordered args, long flags, Python 3. :+1:

**Comment by icodeforyou-dot-net** [comment]
@yuuyins thanks for your work, :rocket:   

I just back on my computer to see you've been working relentlessly on this the past couple of hours! :smile:

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 240549` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>davinci-resolve</li>
  </ul>
</details>

**Review by davidak — APPROVED** [review]
Change looks good

Builds, runs

AMD GPU with OpenCL detected

Screenshot from 2023-06-30 13-41-47

Can someone test a Nvidia GPU with CUDA?

No issue found with function

Tested:

- Create a new project
- Load existing projects
- Playback project timeline
- Cut video
- Apply color correction
- Apply effects
- Apply transition
- Render projects

Thanks for your efforts! :sparkles:

**Review by jshcmpbll — APPROVED** [review]
@BKSalman Nice find on dbus. I was trying update this myself and was hitting a lot of issues. 

Nice work @yuuyins 🥳 Would love help maintaining it if anyone is interested too - im still trying (albeit slowly) to make the derivation build the studio and beta versions too

File: pkgs/applications/video/davinci-resolve/default.nix [hunk 1]
```diff
@@ -73,28 +81,28 @@ let
...
+          --header 'Host: www.blackmagicdesign.com' \
+          --header 'Accept: application/json, text/plain, */*' \
+          --header 'Origin: https://www.blackmagicdesign.com' \
+          --header "$USERAGENT" \
+          --header 'Content-Type: application/json;charset=UTF-8' \
+          --header "Referer: https://www.blackmagicdesign.com/support/download/$REFERID/Linux" \
+          --header 'Accept-Encoding: gzip, deflate, br' \
+          --header 'Accept-Language: en-US,en;q=0.9' \
+          --header 'Authority: www.blackmagicdesign.com' \
+          --header 'Cookie: _ga=GA1.2.1849503966.1518103294; _gid=GA1.2.953840595.1518103294' \
```
Comment (SuperSandro2000) [inline pkgs/applications/video/davinci-resolve/default.nix:94]: You are sure we want to keep and commit these?

File: pkgs/applications/video/davinci-resolve/default.nix [hunk 2]
```diff
@@ -17,30 +18,36 @@
...
-      buildInputs = [ libGLU xorg.libXxf86vm ];
+      buildInputs = [
+        libGLU
+        xorg.libXxf86vm
+      ];
 
       src = runCommandLocal "${pname}-src.zip"
         rec {
           outputHashMode = "recursive";
           outputHashAlgo = "sha256";
```
Comment (SuperSandro2000) [inline pkgs/applications/video/davinci-resolve/default.nix:49]:
```suggestion
```

**Comment by pbsds** [comment]
Considering this package refuses to evaluate on 23.05 due to the python2 dep, i believe a backport can be considered a non-breaking change.

**Comment by BKSalman** [comment]
Thanks for your work @yuuyins and everyone!

**Comment by SimonBrandner** [comment]
Could someone please add the `backport release-23.05` label?

**Comment by pbsds** [comment]
Does it fit the requirements? It's a new major version

**Comment by pbsds** [comment]
I'm much in favor, but won't be the one to call the shot.

**Pull request merged**